src/firefly/js/ui/tap/TapUtil.js

```diff
@@ -255,6 +255,35 @@ async function doLoadTapColumns(serviceUrl, schemaName, tableName) {
     }
 }
 
+
+export const loadTapKeys = memoize(async (serviceUrl) => {
```

Comment:
I don't really like `memoize`, it has the side effect that the cache can be replace for all usages. That always seemed fragile to me. Also, it only uses the first parameter as the cache key (which does not matter in this case). I wrote a `WebUtil` alternative that I use in several places: `memorizeLastCall`. You can try it if you want to otherwise this is fine.